app/routes/wow.shopping-list.tsx

```diff
@@ -0,0 +1,191 @@
+import type { ActionFunction } from '@remix-run/cloudflare'
+import { json } from '@remix-run/cloudflare'
+import { useEffect, useMemo, useState } from 'react'
+import { ContentContainer, PageWrapper, Title } from '~/components/Common'
+import SmallFormContainer from '~/components/form/SmallFormContainer'
+import type { DealItem, WoWDealResponse } from '~/requests/WoW/ShoppingList'
+import WoWShoppingList from '~/requests/WoW/ShoppingList'
+import { getUserSessionData } from '~/sessions'
+import z from 'zod'
+import { useActionData, useNavigation } from '@remix-run/react'
+import { InputWithLabel } from '~/components/form/InputWithLabel'
+import Select from '~/components/form/select'
+import NoResults from '~/components/Common/NoResults'
+import SmallTable from '~/components/WoWResults/FullScan/SmallTable'
+import type { ColumnList } from '~/components/types'
+import ExternalLink from '~/components/utilities/ExternalLink'
+import DebouncedSelectInput from '~/components/Common/DebouncedSelectInput'
+import { items, parseItemsForDataListSelect } from '~/utils/items/id_to_item'
+import { useTypedSelector } from '~/redux/useTypedSelector'
+import { getItemIDByName } from '~/utils/items'
+
+const parseNumber = z.string().transform((value) => parseInt(value, 10))
+
+const validateInput = z.object({
+  itemID: parseNumber,
+  maxPurchasePrice: parseNumber
+})
+
+export const action: ActionFunction = async ({ request }) => {
+  const session = await getUserSessionData(request)
+
+  const region = session.getWoWSessionData().region
+
+  const formData = Object.fromEntries(await request.formData())
+
+  const validatedFormData = validateInput.safeParse(formData)
+  if (!validatedFormData.success) {
+    return json({ exception: 'Invalid Input' })
+  }
+
+  const result = await WoWShoppingList({
+    region,
+    ...validatedFormData.data
+  })
+
+  // await the result and then return the json
+
+  return json({
+    ...(await result.json()),
+    sortby: 'discount'
+  })
+}
+
+type ActionResponseType =
+  | {}
+  | { exception: string }
+  | (WoWDealResponse & { sortby: string })
+
+const ShoppingList = () => {
+  const result = useActionData<ActionResponseType>()
+  const transistion = useNavigation()
+  const { wowItems } = useTypedSelector((state) => state.user)
+  const [itemName, setItemName] = useState<{ name: string; error: string }>({
```

Comment:
Does this need to be an object, can it just be a string? 
You don't appear to be using the error property from it anywhere.